src/filter.rs

```diff
@@ -75,6 +65,15 @@ impl Filter {
             }
           })
       }
+      Self::Rows { limit, step } => {
+        state
+          .matrix()
+          .rows_with_step_mut(0, *limit, *step)
```

Comment:
Here, we can do `nrows.min(state.matrix.height())` to ensure it doesn't panic. (Maybe also need -1 or something.) We should write a test for this.

```suggestion
          .rows_with_step_mut(0, *nrows, *step)
```

Reply:
I think `nrows` in this case, is actually just the amount of rows that operations get applied to, so it would actually be `nrows.min(height / (step + 1))`. I wrote a small test for it in my latest commit.

Reply:
Ohhh, I see. I misunderstood what `rows_with_step_mut` did.

I think this is a bit weird, and users won't understand it. I think that `rows:a:b` should apply to a rows, then skip b rows, then apply to a rows, then skip b rows, etc.